### PR TITLE
Including support to Secrets Manager

### PR DESCRIPTION
**Pull request by renato-rudnicki**
This Pull request adds the option to mount a secret as a volume.

### CODE REVIEW

File: modules/secure-cloud-run-core/variables.tf [hunk 1]
```diff
@@ -151,7 +151,161 @@ variable "env_vars" {
...
+  default     = []
+}
+
+variable "container_command" {
+  description = "Leave blank to use the ENTRYPOINT command defined in the container image, include these only if image entrypoint should be overwritten."
+  type        = list(string)
+  default     = []
+}
+
+variable "env_secret_vars" {
```
Comment (mitchelljamie) [inline modules/secure-cloud-run-core/variables.tf:262]: We should probably not surface the variable to pass in secrets as environmental variables as our security best practice will be to use volume mounts and we want to be very opinionated in this blueprint.

Comment (renato-rudnicki) [reply modules/secure-cloud-run-core/variables.tf]: Good catch! It's fixed now.

**Comment by prabhu34** [comment]
@renato-rudnicki Cloud Run accepts the Volume Mounts for the environment variables that needs to be secure. 
[`env {}` block](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_service#nested_env) should be added for both non-secure and secure env variables. `value_from` field will determine if the value is from a secret.

So if your use case is to create a env variable from secret manager, you should be using `env_secret_vars`, `volumes` and `volume_mounts` variables.

@mitchelljamie Thoughts?

**Comment by mitchelljamie** [comment]
> @renato-rudnicki Cloud Run accepts the Volume Mounts for the environment variables that needs to be secure. [`env {}` block](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_service#nested_env) should be added for both non-secure and secure env variables. `value_from` field will determine if the value is from a secret.
> 
> So if your use case is to create a env variable from secret manager, you should be using `env_secret_vars`, `volumes` and `volume_mounts` variables.
> 
> @mitchelljamie Thoughts?
> 
> Example:
> 
> ```
> # env variable - cleartext value - insecure
>   env_vars = [
>     {
>       name  = "<ENV_VARIABLE_1>"
>       value = "<ENV_VARIABLE_VALUE_1"
>     },
>     {
>       name  = "<ENV_VARIABLE_2>"
>       value = "<ENV_VARIABLE_VALUE_2>"
>     }
>   ]
> # env variable - Value from Secret Manager - Secure
>   env_vars = [
>     {
>       name  = "<ENV_SECRET_VARIABLE_1>"
>       value_from = [{
>         secret_key_ref = {
>           name = "<SECRET_NAME>"
>           key  = "<SECRET_VERSION>"
>         }
>       }]
>     },
>   ]
> # Volume created from GSM Secret 
> volumes = [
>     {
>       name = "<SECRET_VOLUME_NAME>"
>       secret = [{
>         secret_name = "<SECRET_NAME>"
>         items = {
>           path = "<SECRET_PATH>"
>           key  = "<SECRET_VERSION>"
>         }
>       }]
>     },
>   ]
> # Volume Mount added to the Container
> volume_mounts = [
>     {
>       mount_path = "<SECRET_MOUNT_PATH>"
>       name       = "<SECRET_VOLUME_NAME>"
>     },
>   ]
> ```

Hey @prabhu34 - thanks for the input here.  In this blueprint we want to explicitly deny the use of environmental variables to hold secrets as this has been deemed a less secure method than using volume mounts.  We do still want to allow the use of non-secret environmental variables though.  Does that address your concern?

**Review by mitchelljamie — APPROVED** [review]
LGTM

**Pull request merged**